pkg/test/echo/server/forwarder/grpc.go

```diff
@@ -103,15 +102,12 @@ func (c *grpcCall) makeRequest(ctx context.Context, cfg *Config, requestID int)
 			outMD.Set(k, v...)
 		}
 	}
-	outMD.Set("X-Request-Id", strconv.Itoa(requestID))
```

Comment:
This isn't needed and confuses our tests which assume this header is coming from envoy. We don't do this for HTTP